client/src/featureform/resources.py

```diff
@@ -1094,8 +1095,17 @@ class SnowflakeConfig:
     warehouse: str = ""
     role: str = ""
     catalog: Optional[SnowflakeCatalog] = None
+    session_params: Optional[Dict[str, str]] = None
 
     def __post_init__(self):
+
+        if self.session_params:
+            for key, _ in self.session_params.items():
+                if not SnowflakeSessionParamKey.validate_key(key):
```

Comment:
Should this be done in Go? I actually dont mind it in the client, but just curious

Reply:
I waffled on this. I decided to go with a warning instead of an exception to avoid blocking a user down the line in the event Snowflake adds a session param that we don't take into account. I'm happy to move validation to the backend and fail. Thoughts?

Reply:
my thots, since the only way we're interacting with our backend is python client it's probz fine to have validation on the client, but if we were sending this from dxboard as well we'd want backend validation. probz doesn't hurt to keep it here to save a nwork call lol